code/client/munkilib/updatecheck.py

```diff
@@ -2572,7 +2572,10 @@ def getmanifest(partialurl, suppress_errors=False):
     else:
         # request for nested manifest
         manifestdisplayname = partialurl
-        manifestname = os.path.split(partialurl)[1]
+        
+        valid_chars = "-_.() %s%s" % (string.ascii_letters, string.digits)
+        manifestname = ''.join(c if c in valid_chars else "_" for c in partialurl)
+
```

Comment:
Not sure why you are doing this substitution. This could cause the the same problem you are attempting to solve:
manifests/dept1/faculty/printers and manifests/dept1_faculty/printers would be stored as the same name.